code/frameworks/react-vite/src/plugins/react-docgen.ts

```diff
@@ -83,3 +92,24 @@ export function reactDocgen({
     },
   };
 }
+
+export function getReactDocgenImporter(matchPath: TsconfigPaths.MatchPath | undefined) {
+  return makeFsImporter((filename, basedir) => {
+    const mappedFilenameByPaths = (() => {
+      if (matchPath) {
+        const match = matchPath(filename);
+        return match || filename;
+      } else {
+        return filename;
+      }
+    })();
+
+    const result = defaultLookupModule(mappedFilenameByPaths, basedir);
```

Comment:
Can you look into `enhanced-resolve` and see if we could use that over `tsconfig-paths` + `find-up` + `resolve`?
https://www.npmjs.com/package/enhanced-resolve

Reply:
I tried it, but it doesn't have a resolution built-in for TypeScript paths. Here is what I've done: https://github.com/storybookjs/storybook/compare/next...672ccec08f9e8a9e584005580f602c89df66a43d

The plugin is pretty extensible. It allows you to define an `alias` mapping. For this, we would still have to:

1. somehow load the Tsconfig
2. Extract the `paths` mapping
3. Transform it into a format, which `enhanced-resolve` understands (undocumented)

Therefore, I don't see a benefits of using `enhanced-resolve`, since we would still be required to do a lot by hand.

Reply:
Perfect, thanks for the detective work! 🙏